lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -73,8 +73,14 @@ Value RequestCertificateHandler(const MessageOrigin::Ptr& origin, const Dictiona
 		time_t forceRenewalEnd = 1483228800; /* January 1st, 2017 */
 		time_t renewalStart = now + 30 * 24 * 60 * 60;
 
-		if (X509_cmp_time(X509_get_notBefore(cert.get()), &forceRenewalEnd) != -1 && X509_cmp_time(X509_get_notAfter(cert.get()), &renewalStart) != -1) {
+		int days, sec;
+		ASN1_TIME_diff(&days, &sec, X509_get_notBefore(cert.get()), X509_get_notAfter(cert.get()));
 
+		time_t notBefore = 397 * 24 * 60 * 60;
+		time_t notAfter = days * 24 * 60 * 60;
+
+		if ((X509_cmp_time(X509_get_notBefore(cert.get()), &forceRenewalEnd) != -1 && X509_cmp_time(X509_get_notAfter(cert.get()), &renewalStart) != -1)
+			|| notAfter <= notBefore) {
```

Comment:
Why do you put the values into variables and don’t compare them directly?
Why the same multiplication on both sides?

Reply:
To convert the time differences in days to timestamp.

Reply:
What for?